### PR TITLE
add tests for PublicViewer support

### PR DESCRIPTION
**Pull request by filariow**
linked to:
* https://github.com/codeready-toolchain/host-operator/pull/998
* https://github.com/codeready-toolchain/api/pull/410
* https://github.com/codeready-toolchain/toolchain-common/pull/377
* https://github.com/codeready-toolchain/registration-service/pull/417

### CODE REVIEW

File: make/test.mk [hunk 1]
```diff
@@ -116,15 +124,33 @@ test-e2e-registration-local:
 .PHONY: e2e-run-parallel
 e2e-run-parallel:
 	@echo "Running e2e tests in parallel..."
+	oc patch -n ${HOST_NS} toolchainconfigs.toolchain.dev.openshift.com config --patch='{"spec":{"global":{"publicViewer":{"enabled":false}}}}' --type=merge
+	oc delete -n ${HOST_NS} pods -l control-plane=controller-manager
+	oc delete -n ${HOST_NS} pod -l name=registration-service
```
Comment (sadlerap) [inline make/test.mk:129]: Nit: What's the difference between `oc delete pods` (plural) and `oc delete pod` (singular)?  I'd prefer a convention that we stick to.

File: test/e2e/parallel/proxy_test.go [hunk 2]
```diff
@@ -641,7 +625,7 @@ func runWatcher(t *testing.T, awaitilities wait.Awaitilities) *sync.WaitGroup {
 				Get()
 			t.Logf("stopping the watch after %s", time.Since(started))
 
-			require.EqualError(t, err, "unexpected error when reading response body. Please retry. Original error: context deadline exceeded", "The call should be terminated by the context timeout")
+			assert.EqualError(t, err, "unexpected error when reading response body. Please retry. Original error: context deadline exceeded", "The call should be terminated by the context timeout")
```
Comment (sadlerap) [inline test/e2e/parallel/proxy_test.go:628]: Revert this back, and I think checks will pass:

```suggestion
			require.EqualError(t, err, "unexpected error when reading response body. Please retry. Original error: context deadline exceeded", "The call should be terminated by the context timeout")
```

**Review by sadlerap — COMMENTED** [review]
some small nits

File: make/test.mk (hunk 1 above)
Comment (filariow) [reply make/test.mk:129]: a typo 😄 good catch!

**Pull request closed without merging**